pkg/blobstore/replication/noop_blob_replicator.go

```diff
@@ -0,0 +1,31 @@
+package replication
+
+import (
+	"context"
+
+	"github.com/buildbarn/bb-storage/pkg/blobstore"
+	"github.com/buildbarn/bb-storage/pkg/blobstore/buffer"
+	"github.com/buildbarn/bb-storage/pkg/digest"
+)
+
+type noopBlobReplicator struct {
+	source blobstore.BlobAccess
+}
+
+// NewNoopBlobReplicator creates a BlobReplicator that can be used to
+// access a single source without replication.
+//
+// It is useful for the BlobAccess variants where replication is optional.
+func NewNoopBlobReplicator(source blobstore.BlobAccess) BlobReplicator {
+	return &noopBlobReplicator{
```

Comment:
This type is sufficiently small that you don't need to return this by reference. Just do `return noopBlobReplicator{ ... }` and remove the pointer from the receiver types.

Reply:
I don't think it makes any real difference, but done.